File: python/course-repo/solution/05-heartbeats-and-checkpointing/src/webhooks/shared.py
from dataclasses import dataclass, field
from typing import Any

TASK_QUEUE = "webhook-queue"
WEBHOOK_RECEIVER_URL = "http://localhost:9000/hooks"


@dataclass
class WebhookDelivery:
    url: str
    payload: dict
    event_id: str


@dataclass
class WebhookDeliveryBatch:
    url: str
    items: list[dict[str, Any]] = field(default_factory=list)

File: typescript/course-repo/solution/05-heartbeats-and-checkpointing/src/shared.ts
export const TASK_QUEUE = 'webhook-queue';
export const WEBHOOK_RECEIVER_URL = 'http://localhost:9000/hooks';

export interface WebhookDelivery {
  url: string;
  payload: Record<string, unknown>;
  eventId: string;
}

export interface WebhookDeliveryBatch {
  url: string;
  items: Record<string, unknown>[];
}
